File: frontend/src/types/student.ts
export type StudentStatus = 'ACTIVE' | 'INACTIVE' | 'SUSPENDED'
export type Gender = 'MALE' | 'FEMALE' | 'OTHER'

export interface EmergencyContact {
  name: string
  phone: string
  relationship?: string
}

export interface Student {
  id: string
  name: string
  photoUrl?: string | null
  cpf: string
  rg?: string
  gender: Gender
  phone: string
  email: string
  zipCode: string
  address: string
  addressNumber?: string
  neighborhood?: string
  city?: string
  state?: string
  emergencyContact: EmergencyContact
  birthDate: string
  notes?: string
  status: StudentStatus
  registrationCode: string
  createdAt: string
  updatedAt: string
}

export type StudentPayload = Omit<Student, 'id' | 'createdAt' | 'updatedAt' | 'registrationCode'>

File: backend/src/main/java/com/gymflow/pro/entity/Student.java
package com.gymflow.pro.entity;

import com.gymflow.pro.entity.enums.StudentStatus;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "students")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Student {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    @Column(nullable = false, length = 150)
    private String name;

    @Column(name = "photo_url")
    private String photoUrl;

    @Column(nullable = false, length = 14, unique = true)
    private String cpf;

    @Column(length = 20)
    private String rg;

    private String gender;

    @Column(nullable = false, length = 30)
    private String phone;

    @Column(nullable = false, length = 180)
    private String email;

    @Column(name = "zip_code", length = 10)
    private String zipCode;

    private String address;

    @Column(name = "address_number", length = 20)
    private String addressNumber;

    @Column(name = "address_complement")
    private String addressComplement;

    private String neighborhood;

    private String city;

    @Column(length = 2)
    private String state;

    @Column(name = "emergency_contact_name")
    private String emergencyContactName;

    @Column(name = "emergency_contact_phone")
    private String emergencyContactPhone;

    @Column(name = "birth_date", nullable = false)
    private LocalDate birthDate;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @Builder.Default
    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private StudentStatus status = StudentStatus.ACTIVE;

    @Column(name = "registration_code", nullable = false, unique = true, length = 20)
    private String registrationCode;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}
